Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -269,11 +377,21 @@ void TextBox::on_editing_finished() {
 
 void TextBox::on_text_edited(const QString& text) {
   m_current = text;
+  update_placeholder_text();
   m_current_signal(m_current);
 }
 
-void TextBox::elide_text() {
-  auto font_metrics = m_line_edit->fontMetrics();
+void TextBox::on_text_changed(const QString& text) {
```

Comment:
I think this function is no longer needed. QSignalBlocker in the elide_text and update_display_text function can be removed as well. The setText function of the QLineEdit can trigger the textChanged signal. The function will not be called externally, and calling the function internally doesn’t expect to emit the textChanged signal.

Reply:
Signal blocker and on_text_changed have been removed.